docs/book/src/reference/watching-objects.md

```diff
@@ -0,0 +1,21 @@
+# Watching Objects
+
+By default, Kubebuilder and the Controller Runtime libraries allow for controllers
+to easily watch the resources that they manage as well as dependent resources that are `Owned`
+by the controller. However, those are not always the only resources that need to be watched in the cluster.
+
+There are many examples of Specs that have references to external resources.
+- Ingresses have references to Service objects
+- Pods have references to ConfigMaps, Secrets and Volumes
+- Deployments and Services have references to Pods
+
+This same functionality can be added to CRDs and custom controllers.
+This will allow for objects to be reconciled when an object it references is changed.
+
```

Comment:
I think that we would need to add here the controller-runtime docs over the watch as well.